Frameworks/CoreGraphics/CGFont.mm

```diff
@@ -682,69 +213,108 @@ CFDictionaryRef CGFontCopyVariations(CGFontRef font) {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes Returns family name
 */
-CGFontRef CGFontCreateCopyWithVariations(CGFontRef font, CFDictionaryRef variations) {
-    UNIMPLEMENTED();
-    return StubReturn();
+CFStringRef CGFontCopyFullName(CGFontRef font) {
+    if (!font) {
+        return nullptr;
+    }
+    return _DWriteFontCopyInformationalString(font->_dwriteFontFace, DWRITE_INFORMATIONAL_STRING_FULL_NAME);
+}
+
+// Private helper for getting a DWRITE_FONT_METRICS struct, to avoid code duplication
+static struct DWRITE_FONT_METRICS __CGFontGetDWriteMetrics(CGFontRef font) {
+    if (!font->_cachedMetrics) {
+        font->_dwriteFontFace->GetMetrics(&const_cast<struct __CGFont*>(font)->_metrics);
+    }
+    return font->_metrics;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-CFDataRef CGFontCreatePostScriptEncoding(CGFontRef font, const CGGlyph encoding[256]) {
-    UNIMPLEMENTED();
-    return StubReturn();
+int CGFontGetAscent(CGFontRef font) {
+    // Value for CGFont metrics are specified in 'glyph space units', which appear to be the same as DWrite's 'design units'
+    return font ? __CGFontGetDWriteMetrics(font).ascent : 0;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-CFDataRef CGFontCreatePostScriptSubset(CGFontRef font,
-                                       CFStringRef subsetName,
-                                       CGFontPostScriptFormat format,
-                                       const CGGlyph* glyphs,
-                                       size_t count,
-                                       const CGGlyph encoding[256]) {
-    UNIMPLEMENTED();
-    return StubReturn();
+int CGFontGetDescent(CGFontRef font) {
+    // Value for CGFont metrics are specified in 'glyph space units', which appear to be the same as DWrite's 'design units'
+    return font ? __CGFontGetDWriteMetrics(font).descent : 0;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-CGGlyph CGFontGetGlyphWithGlyphName(CGFontRef font, CFStringRef name) {
-    UNIMPLEMENTED();
-    return StubReturn();
+int CGFontGetLeading(CGFontRef font) {
+    // Value for CGFont metrics are specified in 'glyph space units', which appear to be the same as DWrite's 'design units'
+    return font ? __CGFontGetDWriteMetrics(font).lineGap : 0;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+*/
+int CGFontGetCapHeight(CGFontRef font) {
+    // Value for CGFont metrics are specified in 'glyph space units', which appear to be the same as DWrite's 'design units'
+    return font ? __CGFontGetDWriteMetrics(font).capHeight : 0;
+}
+
+/**
+ @Status Interoperable
+*/
+int CGFontGetXHeight(CGFontRef font) {
+    // Value for CGFont metrics are specified in 'glyph space units', which appear to be the same as DWrite's 'design units'
+    return font ? __CGFontGetDWriteMetrics(font).xHeight : 0;
+}
+
+/**
+ @Status Interoperable
+*/
+CGRect CGFontGetFontBBox(CGFontRef font) {
+    if (!font) {
```

Comment:
yes but this keeps it more in line with the functions that do have to do work

---
In reply to: [83672784](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r83672784) [](ancestors = 83672784)